tests/OmniSharp.Roslyn.CSharp.Tests/TypeLookUpV2Facts.cs

```diff
@@ -0,0 +1,326 @@
+        }
+
+        [Fact]
+        public async Task CheckXmlDocumentationNestedTagParamRef()
+        {
+            string content = @"
+public class TestClass
+{
+    /// <summary>Creates a new array of arbitrary type <typeparamref name=""T""/></summary>
+    /// <typeparam name=""T"">The element type of the array</typeparam>
+    public static T[] mk$$Array<T>(int n)
+    {
+        return new T[n];
+    }
+}";
+            var response = await GetTypeLookUpResponse(content);
+            var expected =
+            @"Summary: Creates a new array of arbitrary type T ";
+            Assert.Equal(expected, response.DocComment.SummaryText);
+        }
+
+        [Fact]
+        public async Task CheckXmlDocumentationNestedTagCode()
+        {
+            string content = @"
+public class TestClass
+{
+    /// <example>This sample shows how to call the <see cref=""GetZero""/> method.
+    /// <code>
+    /// class TestClass 
+    /// {
+    ///     static int Main() 
+    ///     {
+    ///         return GetZero();
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    public static int $$GetZero()
+    {
+        return 0;
+    }
+}";
+            var response = await GetTypeLookUpResponse(content);
+            var expected =
+            @"Example: This sample shows how to call the TestClass.GetZero method.
+
+    class TestClass 
+    {
+        static int Main() 
+        {
+            return GetZero();
+        }
+    }
+    ";
+            Assert.Equal(expected.Replace("\r",""), response.DocComment.ExampleText);
+        }
+
+        [Fact]
+        public async Task CheckXmlDocumentationNestedTagPara()
+        {
+            string content = @"
+public class TestClass
+{
+    /// <summary>DoWork is a method in the TestClass class.
+    /// <para>Here's how you could make a second paragraph in a description.</para>
+    /// </summary>
+    public static void Do$$Work(int Int1)
+    {
+    }
+}
+            ";
+            var response = await GetTypeLookUpResponse(content);
+            var expected =
+@"Summary: DoWork is a method in the TestClass class.
+
+Here's how you could make a second paragraph in a description.";
+            Assert.Equal(expected.Replace("\r", ""), response.DocComment.SummaryText);
+        }
+
+        [Fact]
+        public async Task CheckXmlDocumentationNestedTagSeeAlso()
+        {
+            string content = @"
+public class TestClass
+{
+    /// <summary>DoWork is a method in the TestClass class.
+    /// <seealso cref=""TestClass.Main""/>
+    /// </summary>
+            public static void Do$$Work(int Int1)
+            {
+            }
+
+            static void Main()
+            {
+            }
+}";
+        var response = await GetTypeLookUpResponse(content);
+            var expected =
+@"Summary: DoWork is a method in the TestClass class.
+See also: TestClass.Main ";
+            Assert.Equal(expected.Replace("\r",""), response.DocComment.SummaryText);
+        }
+
+    }
```

Comment:
I like that you have tests for each type of section or tag, but there aren't many tests that validate that multiple sections all show up correctly together? Could you add some tests that verify what happens if you have a <summary>, <params> and <returns> all in one doc comment (for example).

Can you also add tests that validate the behavior if <params> elements are interleaved with other elements? eg <param>... <typeparam> ... <param> ...